routers/path_filter.go

```diff
@@ -0,0 +1,26 @@
+package routers
+
+import (
+	"net/http"
+	"path"
+	"strings"
+
+	"github.com/beego/beego/context"
+)
+
+func PathFilter(ctx *context.Context) {
+	urlPath := ctx.Request.URL.Path
+	if !strings.HasPrefix(urlPath, "/api/") && !strings.HasPrefix(urlPath, "/.well-known/") &&
+		!strings.HasPrefix(urlPath, "/cas") && !strings.HasSuffix(urlPath, "/serviceValidate") &&
+		!strings.HasSuffix(urlPath, "/proxy") && !strings.HasSuffix(urlPath, "/proxyValidate") &&
+		!strings.HasSuffix(urlPath, "/validate") && !strings.HasSuffix(urlPath, "/p3/serviceValidate") &&
+		!strings.HasSuffix(urlPath, "/p3/proxyValidate") && !strings.HasSuffix(urlPath, "/samlValidate") {
+		return
```

Comment:
can we simplify it by something like this ?

```
	prefixWhitelist := []string{"/api/", "/.well-known/"}
	suffixWhitelist := []string{"/validate", "/p3/serviceValidate"}

	valid = false 

	for _, prefix := range prefixWhitelist {
		if strings.HasPrefix(urlPath, prefix) {
			valid = true 
			return
		}
	}

	for _, prefix := range suffixWhitelist {
		if strings.HasSuffix(urlPath, prefix) {
			valid = true 
			return
		}
	}
```

Reply:
done